src/lib/snark_work_lib/sub_zkapp_spec.mli

```diff
@@ -11,13 +18,18 @@ module Stable : sig
           ; witness :
               Transaction_snark.Zkapp_command_segment.Witness.Stable.V1.t
           ; spec : Transaction_snark.Zkapp_command_segment.Basic.Stable.V1.t
+          ; which_segment : int
           }
       | Merge of
           { proof1 : Ledger_proof.Stable.V2.t
           ; proof2 : Ledger_proof.Stable.V2.t
+          ; first_segment_of_proof1 : int
+          ; last_segment_of_proof2 : int
           }
     [@@deriving sexp, yojson]
```

Comment:
I'm aware you don't have to send these numbers over network. But I'm just keeping them here.

Could get rid of them in subsequent PRs